java/dataset/src/main/java/org/apache/arrow/dataset/file/NativeScannerAdaptorImpl.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.dataset.file;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import org.apache.arrow.c.ArrowArray;
+import org.apache.arrow.c.Data;
+import org.apache.arrow.dataset.jni.NativeRecordBatchIterator;
+import org.apache.arrow.dataset.scanner.ScanTask;
+import org.apache.arrow.dataset.scanner.Scanner;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.ipc.ArrowReader;
+
+/**
+ * Default implementation of {@link NativeScannerAdaptor}.
+ */
+public class NativeScannerAdaptorImpl implements NativeScannerAdaptor, AutoCloseable {
```

Comment:
Is it needed to create this class?

`IteratorImpl` seems to already handle everything.